extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/GZipTest.java

```diff
@@ -0,0 +1,67 @@
+package io.quarkus.resteasy.reactive;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.vertx.ext.web.Router;
+
+public class GZipTest {
+
+    private static final String APP_PROPS = "" +
+            "quarkus.http.enable-compression=true\n";
+
+    static String longString;
+    static {
+
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < 1000; ++i) {
+            sb.append("Hello RESTEasy Reactive;");
+        }
+        longString = sb.toString();
+    }
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource(new StringAsset(APP_PROPS), "application.properties")
+                    .addClasses(BeanRegisteringRouteUsingObserves.class));
+
+    @Test
+    public void test() throws Exception {
+
+        RestAssured.given().get("/compression").then().statusCode(200)
+                .header("content-encoding", "gzip")
+                .header("content-length", Matchers.not(Matchers.equalTo(Integer.toString(longString.length()))))
+                .body(Matchers.equalTo(longString));
+
+        RestAssured.given().get("/nocompression").then().statusCode(200)
+                .header("content-encoding", "identity")
+                .header("content-length", Matchers.equalTo(Integer.toString(longString.length())))
+                .body(Matchers.equalTo(longString));
+    }
+
+    @ApplicationScoped
+    static class BeanRegisteringRouteUsingObserves {
+
+        public void register(@Observes Router router) {
+
+            router.route("/compression").handler(rc -> {
```

Comment:
This is testing reactive routes, but we want to be testing RESTEasy Reactive :)

Reply:
From available articles (https://quarkus.io/guides/reactive-routes) I got the reactive route part, I tried searching for resources around testing RR but didn't get. Maybe we are supposed to do something like ?https://github.com/quarkusio/quarkus/blob/85d574ec82b3bc4735ed5933abe6cfaecad6b1d7/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/resource/basic/SubResourceRequestFilterTest.java#L61

Reply:
oh wait, perhaps I should be testing this against some RR endpoint

Reply:
Yes, indeed.